src/Offer/CommandHandlers/ImportLabelsHandler.php

```diff
@@ -37,20 +49,88 @@ public function handle($command): void
             return;
         }
 
-        /** @var Udb3ModelLabel $importLabel */
-        foreach ($command->getLabelsToImport() as $importLabel) {
+        $offer = $this->offerRepository->load($command->getItemId());
+
+        $labelsToImport = $command->getLabels();
+        $labelsToImport = $this->fixVisibility($labelsToImport);
+
+        $labelsOnOffer = new Labels();
+        foreach ($offer->getLabels()->asArray() as $labelOnOffer) {
+            $labelsOnOffer = $labelsOnOffer->with(
+                new Label(
+                    new LabelName($labelOnOffer->getName()->toNative()),
+                    $labelOnOffer->isVisible()
+                )
+            );
+        }
+
+        $labelsToKeepOnOffer = $this->lockedLabelRepository->getLockedLabelsForItem($command->getItemId());
```

Comment:
Would there be a way to also get this from the aggregate one way or another? I think this "locked labels" read model is just based on whether labels are added via `LabelAdded` or `LabelsImported`. So maybe we can also store that in the aggregate state and get it from there, and then remove this read model later (when we also remove `ImportLabelVisibilityRequestBodyParser` and it is no longer used.)

Reply:
Will be picked up in https://jira.uitdatabank.be/browse/III-4620